test/integration/ferc1_eia_train_test.py

```diff
@@ -0,0 +1,116 @@
+"""Tests for the validation functions in the ferc1_eia_train module."""
+
+import importlib.resources
+import logging
+
+import pandas as pd
+import pytest
+
+from pudl.analysis.ferc1_eia import restrict_train_connections_on_date_range
+from pudl.analysis.ferc1_eia_train import validate_override_fixes
+from pudl.output.pudltabl import PudlTabl
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="module")
+def pudl_out(pudl_engine, pudl_datastore_fixture):
+    """A PUDL output object for use in CI."""
+    return PudlTabl(
+        pudl_engine,
+        ds=pudl_datastore_fixture,
+        freq="AS",
+        fill_tech_desc=False,
+        fill_net_gen=False,
+    )
+
+
+@pytest.fixture(scope="module")
+def utils_eia860(pudl_out):
+    """The utils_eia860 output table."""
+    return pudl_out.utils_eia860()
+
+
+@pytest.fixture(scope="module")
+def plant_part_list(pudl_out):
+    """The plant_parts_eia output table."""
+    return pudl_out.plant_parts_eia().reset_index()
+
+
+@pytest.fixture(scope="module")
+def ferc1_eia(pudl_out):
+    """The ferc1_eia output table."""
+    return pudl_out.ferc1_eia()
+
+
+@pytest.fixture(scope="module")
+def ferc1_eia_training_data():
+    """The training data for the ferc1_eia matching."""
+    return pd.read_csv(
+        importlib.resources.path("pudl.package_data.glue", "ferc1_eia_train.csv")
+    )
+
+
+@pytest.mark.parametrize(
+    "verified,report_year,record_id_eia_override_1,record_id_ferc1",
+    [
+        pytest.param(
+            ["True"],
+            [2020],
+            ["4270_2020_plant_total_11241"],
+            ["f1_steam_2020_12_454_0_3"],
+        ),
+        # This param fails for the wrong reason -- because the 2019 data isn't available.
+        pytest.param(
+            ["True"],
+            [2019],
+            ["4270_2020_plant_total_11241"],
+            ["f1_steam_2020_12_454_0_3"],
+            marks=pytest.mark.xfail(
+                reason="EIA record year doesn't match FERC record year"
+            ),
```

Comment:
why are you adding the report_year here at all into this test? isn't the report year gleaned from either the `ppe` or the ferc plants table?

Reply:
this report_year column is how the validation function determines whether an EIA record was matched to a FERC record from a different year. Because the report_year column in the override spreadsheet comes from FERC.